### PR TITLE
🐛 Fix FixedLayer attribute sync for amp-bind

### PR DESCRIPTION
**Pull request by jridgewell**
go/ampe/CPjot5epg8-i6QE

### CODE REVIEW

File: test/functional/test-fixed-layer.js [hunk 1]
```diff
@@ -176,6 +176,31 @@ describes.sandboxed('FixedLayer', {}, () => {
     };
   });
 
+  class Attributes {
```
Comment (choumx) [inline test/functional/test-fixed-layer.js]: FakeAttributes

File: test/functional/test-fixed-layer.js [hunk 2]
```diff
@@ -176,6 +176,31 @@ describes.sandboxed('FixedLayer', {}, () => {
...
+            this[i].value = value;
+            return;
+          }
+        }
+        this.push(new Attr(name, value));
+      };
+      return attrs;
+    }
+  }
+  class Attr {
```
Comment (choumx) [inline test/functional/test-fixed-layer.js]: FakeAttr

File: test/functional/test-fixed-layer.js [hunk 3]
```diff
@@ -1512,6 +1537,24 @@ describes.sandboxed('FixedLayer', {}, () => {
       expect(layer.getAttribute('test1')).to.equal(null);
       expect(layer.getAttribute('test2')).to.equal(null);
     });
+
+    it('should sync invalid-named attributes to layer', () => {
+      // Holy poop it's hard to inject an invalid-named attribute.
```
Comment (choumx) [inline test/functional/test-fixed-layer.js]: I also used innerHTML for test-bind-impl.js.

File: test/functional/test-fixed-layer.js [hunk 4]
```diff
@@ -1512,6 +1537,24 @@ describes.sandboxed('FixedLayer', {}, () => {
...
+      expect(attr.name).to.equal('[class]');
+      expect(attr.value).to.equal('amp-bind');
+
+      const body = document.createElement('body');
+      body.attributes.setNamedItem(attr.cloneNode(false));
+
+      expect(div.getAttribute('[class]')).to.equal('amp-bind');
+      expect(body.getAttribute('[class]')).to.equal('amp-bind');
+      expect(body.attributes.getNamedItem('[class]')).to.not.equal(
+          div.attributes.getNamedItem('[class]'));
```
Comment (choumx) [inline test/functional/test-fixed-layer.js]: Shouldn't we be calling `fixedLayer.update()` and testing `layer.getAttribute(...)`?

Comment (jridgewell) [reply test/functional/test-fixed-layer.js]: This is a sanity check test, it's too hard to wire a real DOM node through fixed layer (it'll take a complete refactor of this test file).

Comment (choumx) [reply test/functional/test-fixed-layer.js]: We should at least ensure that `setNamedItem` is called. What if you mimic browser behavior by throwing on illegal chars in your fake `setAttribute()`?

**Pull request merged**